### PR TITLE
Assetto Corsa

### PR DESCRIPTION
**Pull request by Sparten**
Latest update. I guess that it for me for tonight, gonna go race in fullscreen now, for a change.

### CODE REVIEW

File: CrewChiefEx/CrewChiefEx.py [hunk 1]
```diff
@@ -128,23 +138,46 @@ def acMain(ac_version):
 
 def acUpdate(deltaT):
     global siminfo
-    global l_lapcount, lapcount,l_driver,l_drivers,maxSlotId,l_flag
-    updateSharedMemory()
+    global l_lapcount, lapcount,l_driver,l_drivers,maxSlotId,l_flag, timer1
+    
+    timer1 += deltaT
+    
+    if timer1 > 0.050:
```
Comment (mrbelowski) [inline CrewChiefEx/CrewChiefEx.py]: should all the logic below this line be in the 'if' block? Perhaps i'm reading it wrong but it looks like you only call updateSharedMemory() if time1 > 0.05, but you call siminfo.getSharedmem() and all the other stuff (including splineToDistanceRoundTrack) even if timer1 <=0.05. Do we need to do all this other stuff ever time AC calls acUpdate(deltaT), or should we only do it if timer1 > 0.05?

File: CrewChiefV4/ACS/ACSGameStateMapper.cs [hunk 2]
```diff
@@ -413,33 +495,49 @@ public GameStateData mapToGameStateData(Object memoryMappedFileStruct, GameState
...
-                    if (currentGameState.SessionData.SectorNumber == 3)
+                    if (currentGameState.SessionData.SectorNumber == 3 )
                     {
-                        
-                        //Console.WriteLine("sector2 time:{0:D2}m:{1:D2}s:{2:D3}ms", ts.Minutes, ts.Seconds, ts.Milliseconds);
+
+
                         currentGameState.SessionData.SessionTimesAtEndOfSectors[2] = currentGameState.SessionData.SessionRunningTime;
+                        if (isOnline)
+                            sectortime = addPlayerLapdata(sectortime, currentGameState.SessionData.SessionRunningTime);
```
Comment (mrbelowski) [inline CrewChiefV4/ACS/ACSGameStateMapper.cs]: I know it's a minor thing, but please use curly braces for all if statements, even if it's only a 1-liner:
if (isOnline) 
{
   sectortime = ....
}

**Comment by mrbelowski** [comment]
Might be worth tidying up your C# code a little here - it might be GitHub's web interface, but the indentation looks a bit inconsistent and 'if' statements should always use curly braces - 
if (something)
{
   someCodeEvenIfItsOnlyOneLine
}

I added a comment to the python stuff too - i wondered if it was still doing too much work on every tick.

**Comment by Sparten** [comment]
Your not silly at all cause your right, and i will do my best to follow your directions, so i will clean up and then commit again.

File: CrewChiefV4/ACS/ACSGameStateMapper.cs (hunk 2 above)
Comment (Sparten) [inline CrewChiefV4/ACS/ACSGameStateMapper.cs]: No problem I will make sure to do so.

File: CrewChiefEx/CrewChiefEx.py (hunk 1 above)
Comment (Sparten) [inline CrewChiefEx/CrewChiefEx.py]: I will comment out all the "debug"  below the 'if' timer, its really only for developing and checking values, if that ok

**Comment by Sparten** [comment]
Cleaned it up a bit I hope i got all the curly braces aswell.

**Comment by Sparten** [comment]
I cleaned up the python as much as i could, i will keep it like that for releases from now on.

**Comment by mrbelowski** [comment]
Oops, I just noticed that the stuff in CrewChiefEx.py after the if time1 > 0.05 check was commented out anyway - sorry about that :)

Will merge now

**Comment by Sparten** [comment]
I added spotter car length to user settings. I hope you dont mind.

**Pull request merged**